lib/Onnxifi/onnxifiGlow.cpp

```diff
@@ -245,6 +245,23 @@ onnxWaitEvent(onnxEvent event) {
   return ONNXIFI_STATUS_SUCCESS;
 }
 
+/// Query ONNXIFI event state without blocking.
+ONNXIFI_PUBLIC ONNXIFI_CHECK_RESULT onnxStatus ONNXIFI_ABI
+  onnxGetEventState(
+    onnxEvent event,
+    onnxEventState* state) {
+  auto *glowEvent = reinterpret_cast<glow::onnxifi::EventPtr>(event);
+  if (!glowEvent) {
+    return ONNXIFI_STATUS_INVALID_EVENT;
+  }
+  if (!state) {
```

Comment:
This function must set `*state` to `ONNXIFI_EVENT_STATE_INVALID` if it fails. I suggest that you do the check `state != nullptr`, and immediately set `*state` to `ONNXIFI_EVENT_STATE_INVALID` if the check succeeds. You will later overwrite `*state` with the proper value before returning.

Reply:
@Maratyszcza good point!